### ISSUE TITLE
Config overrides do not work for files in the parent directory

### ISSUE THREAD

**Issue by Shayan-To** [prettier/prettier#13556]
I'm using prettier to format some json files with custom extensions (like VSCode's `.code-workspace` files). The files are in the parent directory of the project, so I'm running `prettier --write ../**/*.{json-1,json-2}`.

The problem I have is that I cannot tell prettier that parser for `.json-1` files is `"json"`. This is my config:

```json
{
    "overrides": [
        {
            "files": "*.json-1",
            "options": {
                "parser": "json"
            }
        }
    ]
}
```

And when I run prettier, it gives me this output:

```
[error] No parser could be inferred for file: ../test.json-1
```

I have also tried changing `"files": "*.json-1"` to `"files": "**/*.json-1"` and `"files": "../**/*.json-1"`, but the behavior is the same.

**Environments:**

-    Prettier Version: 2.7.1
-    Running Prettier via: CLI
-    Runtime: Node.js v16.14.2
-    Operating System: Ubuntu 20.04.4
-    Prettier plugins (if any): --

**Steps to reproduce:**

Here's the repro (in the `prettier-parent-overrides` branch): https://github.com/Shayan-To/repros/tree/prettier-parent-overrides

Go into the project directory `cd prettier-proj`, and after `npm install`, run `npm run repro`.

There are files both in the parent directory, and in the project directory. Prettier can infer the parser for none of the files in the parent directory.

**Expected behavior:**

The `overrides` configurations for the files in the parent directory can be set.

**Actual behavior:**

There seems to be no way to set `overrides` configurations for the files in the parent directory.

**Comment by thorn0** [comment]
This is by design. Config files are resolved per file starting from the file's directory and up the tree. The current working directory doesn't participate in that process. It might probably be convenient to do what you want sometimes, but this use case doesn't look widespread enough to justify a huge change in how configs are resolved, sorry.

**Comment by Shayan-To** [comment]
@thorn0 I don't think I completely understand. Let me ask some questions.

You say the configs are per file. So you mean that I can add a prettierrc to a sub-directory of my project, and it will apply to files in that sub-directory?

What about files that are not inside the project directory? What if I'm formatting files in the parent directory (`prettier --write ../**/*.json`)? Will the prettierrc from the project directory apply there? Will the prettierrc from the parent directory apply there? How about editorconfig files in the project/parent directories?

And I don't understand why you are supporting reading configurations from `.editorconfig` files, but are not conforming to their standard. They say explicitly in their website that if a `.editorconfig` file is not `root=true`, then you should look in the parent directories for one with `root=true`.

I have a main project, and my node project is in a sub-directory. And by editorconfig's standard, I should be able to have my `.editorconfig` file in the main project directory (so in parent directory of prettier's project), and have its config applied to the whole directory and its sub-directories. Except, prettier does not respect that.

**Comment by thorn0** [comment]
>  you mean that I can add a prettierrc to a sub-directory of my project, and it will apply to files in that sub-directory?

Right

> What if I'm formatting files in the parent directory (prettier --write ../**/*.json)? Will the prettierrc from the project directory apply there?

Only for files inside the project directory because your glob includes them.

> Will the prettierrc from the parent directory apply there?

Yes

As for editorconfig, AFAIK, it's all intended too, but I can't comment more on it now. Feel free to search the issue tracker yourself to find the discussions that lead to the current state of things.

**Comment by thorn0** [comment]
Forgot about this: https://prettier.io/docs/en/cli.html#--find-config-path-and---config
You can specify the config file in the command line to get what you need.

**Comment by Shayan-To** [comment]
@thorn0 Thanks. Setting the config file explicitly worked very well for my use case. To provide some feedback...

I first moved my `.prettierrc.json` file to the parent folder, so it applies to all the files. Then I noticed the inconsistent behavior between `.prettierrc.json` files, and `.prettierignore` files (`.prettierignore` file must be in the project directory, and it will apply everywhere). Because of this inconsistency, I decided to put everything in the project directory, and use your suggestion and set the config file explicitly by `--config` switch.
